### PR TITLE
Match all

### PR DESCRIPTION
**Pull request by sorribas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,15 @@ Arguments:
...
 first match. If no pattern matches `null` is returned.
 
+### `patterns.matchAll(target)`
+
+Arguments
+
+- `target` - The string that should be matched against each pattern
+
+Runs through each pattern in order and returns an array of match objects.
+If no patter matches, an empty array is returned.
```
Comment (watson) [inline README.md]: Minor typo: s/patter/pattern/

File: test.js [hunk 2]
```diff
@@ -72,3 +72,17 @@ test('last next function', function (t) {
...
+
+test('matchAll', function (t) {
+  var p = Patterns();
+  p.add(/foo/, 1);
+  p.add(/foo/, 2);
+  p.add(/foo/, 3);
+  var matches = p.matchAll('foo');
+  t.deepEqual(matches[0].value, 1);
+  t.deepEqual(matches[1].value, 2);
+  t.deepEqual(matches[2].value, 3);
```
Comment (watson) [inline test.js]: Why not just use `t.equal` instead of `t.deepEqual`?

**Comment by watson** [comment]
@sorribas Nice feature :) Thanks!

**Comment by sorribas** [comment]
@watson fixed.

**Pull request merged**